tonic/models/vic/grid_params.py

```diff
@@ -1015,18 +1415,23 @@ def veg(veg_file, soil_dict, max_roots=3, veg_classes=11,
         sigma_slope = np.zeros((cells, veg_classes))
         lag_one = np.zeros((cells, veg_classes))
         fetch = np.zeros((cells, veg_classes))
-    if lai_index:
-        lfactor = 2
+    lfactor = 1
+    if vegparam_lai:
+        lfactor += 1
         lai = np.zeros((cells, veg_classes, 12))
-    else:
-        lfactor = 1
+    if vegparam_fcan:
+        lfactor += 1
+        fcan = np.zeros((cells, veg_classes, 12))
+    if vegparam_albedo:
+        lfactor += 1
+        albedo = np.zeros((cells, veg_classes, 12))
```

Comment:
Maybe we should swap out these `12`s for `MONTHS_PER_YEAR`.  I think that value is set in tonic.share or you could put it in the header of this file.